### ISSUE TITLE
Multi-project configuration doesn't type check files not included in any project

### ISSUE THREAD

**Issue by ark120202** [typescript-eslint/typescript-eslint#724]
**What code were you trying to parse?**

```ts
[1].forEach(value => value + 1);
```

**What did you expect to happen?**

No warnings

**What actually happened?**

A warning from `@typescript-eslint/restrict-plus-operands` rule.

After a bit of debugging I found out that type checker returned `any` type for the `value` node. It happened because `project` didn't contain a reference to package's tsconfig file, so this file was linted in a context of an inferred project.

I've tried to reproduce it in an isolated environment and looks like it works correctly if there is only one project.

**UPD:** After reading source it makes sense that it can't do the same thing it does with a single config, but falling back to the same program used for non-type-checked rules should also make rules handle it like it does without any project specified - throw an error.

**Versions**

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/typescript-estree` | `1.12.0` |
| `TypeScript`                           | `3.5.3` |
| `node`                                 | `12.1.0` |
| `npm`                                  | `6.9.0` |

**Comment by bradzacher** [comment]
this is due to the same reason that the plugin is slow when you don't include all files within a tsconfig. It essentially handles it by fudging a lot.

The parser doesn't have a context of partially typechecked or not per file, it just reports that every file has either been fully typechecked (i.e. `project` is provided), or it's not.

We're planning on fixing this by throwing hard errors if the file isn't within the specified project (which is the same thing tslint did).
